components/movementsensor/adxl345/adxl345.go

```diff
@@ -81,9 +102,32 @@ func init() {
 }
 
 type adxl345 struct {
-	bus        board.I2C
-	i2cAddress byte
-	logger     golog.Logger
+	bus               board.I2C
+	i2cAddress        byte
+	logger            golog.Logger
+	echoInterrupt     board.DigitalInterrupt
+	interruptsEnabled []string
+	ticksChan         chan board.Tick
+	interruptsFound   []string
```

Comment:
Discussion question: do we also want to store the timestamp at which each interrupt is encountered? I think something similar to that should be available from the `Tick` struct, though Gautham would know best.

Reply:
I think it would be possible the only confusion would be cases where multiple interrupts have occurred since the last time the `INT_SOURCE` register has been read, so we would have an invalid time stamp for all but one of those interrupts. 

Having said this, the goroutine is continuously monitoring the state of the pin and reading from the register if it changes so this shouldn't really happen - just a thought to consider